### DISCUSSION TITLE
Azure OpenAI API

### DISCUSSION THREAD

**Discussion by liuba1223** [logancyang/obsidian-copilot#100] [Q&A]
After using the Azure OpenAI API, I found that the QA: active note feature is no longer functioning properly. I attempted to use the CoheareAI API as an embedding provider, but it still did not work.

**Comment by logancyang** [comment]
Can you turn on debug mode in settings and share a screenshot of your console? Pls also share how you reproduce this issue step by step so I can reproduce it myself.

**Comment by liuba1223** [comment]
---
*** DEBUG INFO ***user message: Summarize this article.model: gpt-35-turbotemperature: 0.2maxTokens: 8000system message: You are Obsidian Copilot, a helpful assistant thatintegrates AI to Obsidian note-taking.chat context turns: 6
plugin:copilot:64583
chain:ConversationChain
Chat memory:BufferWindowMemory
plugin:copilot:55554
plugin:copilot:55555
DeyTools failed to load source map: Could not load content for app://obsioian.md/main.js.map: System error: net::ERR FILE NOT FOUND
---
Under normal circumstances, when I activate "QA:active note", the prompt "Vector store created successfully" should appear. However, after I switched to Azure API, this prompt no longer appears and active note is not functioning either.

**Reply by logancyang** [reply]
Can you post a screenshot, the message I'm looking for should be below all this. I suspect there are some API error messages where you failed to call those API either because of missing or wrong credentials.

**Reply by liuba1223** [reply]
Uncaught (in promise) Error: (Azure) OpenAI API key not found
    at new OpenAIEmbeddings (plugin:copilot:54492:13)
    at AIState.getEmbeddingsAPI (plugin:copilot:55332:33)
    at AIState.buildIndex (plugin:copilot:55482:32)
    at async AIState.setChain (plugin:copilot:55451:11)

Is this it? I'm sorry, but I failed to upload the picture.How do you normally use the Azure API?

**Reply by liuba1223** [reply]
Uncaught (in promise) [ERROR in Core Search Assistant] failed to find the search input form.
plugin:copilot:59669 Uncaught (in promise) Error: OpenAI or Azure OpenAI API key not found
    at new OpenAIEmbeddings (plugin:copilot:59669:13)
    at AIState.getEmbeddingsAPI (plugin:copilot:60537:33)
    at AIState.buildIndex (plugin:copilot:60696:32)
    at async AIState.setChain (plugin:copilot:60665:11)
plugin:copilot:69799 *** DEBUG INFO ***
user message: 搜索
model: gpt-35-turbo
temperature: 0.2
maxTokens: 8000
system message: You are Obsidian Copilot, a helpful assistant that integrates AI to Obsidian note-taking.
chat context turns: 6
plugin:copilot:60768 chain: 
ConversationChain
plugin:copilot:60769 Chat memory: 
BufferWindowMemory
DevTools failed to load source map: Could not load content for app://obsidian.md/main.js.map: System error: net::ERR_FILE_NOT_FOUND

**Comment by liuba1223** [comment]
I have updated the latest version of the plugin and filled in the corresponding information as instructed. However, QA:active note still doesn't work. Could you please help me check where I made a mistake in the settings?
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "logancyang/obsidian-copilot"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月11日(星期二) 上午9:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [logancyang/obsidian-copilot] Azure OpenAI API (Discussion #100)





 
Can you post a screenshot, the message I'm looking for should be below all this. I suspect there are some API error messages where you failed to call those API either because of missing or wrong credentials.

**Reply by logancyang** [reply]
I can't see your settings. There's not much I can do to help debug your deployment. One helpful thing I tried was Azure's support chat, they are quite helpful, just let them help you figure out the right params, first make sure using `curl` works, and you will be able to use this plugin without an issue.

But I see Azure params can be hard to get right at first. Will put the curl command example in the FAQ.

**Reply by liuba1223** [reply]
Thank you very much for your patient explanation.
